### PR TITLE
Fix build errors and use proper includes

### PR DESCRIPTION
**Pull request by rbx**

### CODE REVIEW

**Comment by ktf** [comment]
How come you have the `include/options`, `include/zeromq` directories? I do not see them.

**Comment by MohammadAlTurany** [comment]
You have to wait for the merge of PR 517 in FairRoot, Sorry for that!!

File: CCDB/include/CCDB/Backend.h [hunk 1]
```diff
@@ -8,7 +8,7 @@
 #include <string>
 #include <vector>
 
-#include "FairMQDevice.h"
+#include <FairMQDevice.h>
```
Comment (sawenzel) [inline CCDB/include/CCDB/Backend.h]: why is this necessary?

File: Utilities/DataFlow/src/TimeframeValidatorDevice.cxx [hunk 2]
```diff
@@ -9,7 +9,7 @@
 #include "DataFlow/TimeframeValidatorDevice.h"
 #include "Headers/SubframeMetadata.h"
 #include "Headers/DataHeader.h"
-#include "FairMQProgOptions.h"
+#include <options/FairMQProgOptions.h>
```
Comment (sawenzel) [inline Utilities/DataFlow/src/TimeframeValidatorDevice.cxx]: this directory does not exist in the installation

Comment (MohammadAlTurany) [inline Utilities/DataFlow/src/TimeframeValidatorDevice.cxx]: Now in dev it does

File: CCDB/include/CCDB/Backend.h (hunk 1 above)
Comment (dennisklein) [inline CCDB/include/CCDB/Backend.h]: https://gcc.gnu.org/onlinedocs/cpp/Include-Syntax.html

It is just cosmetics here, because the quoted syntax did a fallback to `<>` anyways. You want to use the quoted syntax only for files included from your own source tree.

**Comment by ktf** [comment]
@MohammadAlTurany I am now splitting the tests in two:

- Using the current hash for dev
- Using the tip of the dev branch

It would be good if (after easter, I am leaving today) we could setup the PR tests for FairRoot which checks that the dev of O2 works with the dev of FairRoot, so that we avoid this kind of issues in the future.

**Comment by ktf** [comment]
You still seem to need https://github.com/FairRootGroup/FairRoot/pull/516, because it's not a given that zeromq will be installed in your system include path.

File: CCDB/include/CCDB/Backend.h (hunk 1 above)
Comment (sawenzel) [inline CCDB/include/CCDB/Backend.h]: As you say ... it's cosmetics

File: cmake/O2Dependencies.cmake [hunk 3]
```diff
@@ -162,6 +161,7 @@ o2_define_bucket(
 
     INCLUDE_DIRECTORIES
     ${FAIRROOT_INCLUDE_DIR}
+    ${FAIRROOT_INCLUDE_DIR}/fairmq # temporary fix, until bucket system works with imported targets
```
Comment (dennisklein) [inline cmake/O2Dependencies.cmake]: The `dev` branch of FairRoot now exports and installs all FairMQ targets (which also get installed by `make install`, so without unit tests) in a file `${FAIRROOTPATH}/include/cmake/FairMQ.cmake`. The AliceO2 build system calls `find_package(FairRoot)` somewhere which executes the cmake script `${FAIRROOTPATH}/share/fairbase/modules/FindFairRoot.cmake` which [looks](https://github.com/FairRootGroup/FairRoot/blob/dev/cmake/modules/FindFairRoot.cmake#L45-L52) for the exported FairMQ targets.

This has the benefit that link dependencies and include directories are shipped together with the exported/imported target. The file `${FAIRROOTPATH}/include/cmake/FairMQ.cmake` contains e.g.
```cmake
(...)
# Create imported target FairRoot::FairMQ
add_library(FairRoot::FairMQ SHARED IMPORTED)

set_target_properties(FairRoot::FairMQ PROPERTIES
  INTERFACE_INCLUDE_DIRECTORIES "${_IMPORT_PREFIX}/include/fairmq"
  INTERFACE_LINK_LIBRARIES "dl;pthread;Boost::boost;Boost::log;Boost::log_setup;Boost::program_options;Boost::thread;Boost::system;Boost::filesystem;Boost::regex;Boost::date_time"
)
(...)
```
As you can see from this PR you can simplify the dependency management quite a lot and give proper control over path configurations to the upstream team. In a future patch FairRoot might also export the non-FairMQ targets in this way - FairMQ is a first test to see, if it is useful and works.

In the case where we generate the `rootcint` command in the function `O2_ROOT_GENERATE_DICTIONARY`, we have to generate the `-I` options ourselves. Currently, the bucket dependency management does not know about [target properties](https://cmake.org/cmake/help/v3.0/manual/cmake-properties.7.html#properties-on-targets) and therefore does not generate the correct `-I` option from the imported `FairRoot::FairMQ` target property `INTERFACE_INCLUDE_DIRECTORIES`. That is the reason for the "hack" in line 164 above.

The bucket dependency resolver (cmake function `GET_BUCKET_CONTENT`) can probably be extended so that it looks up the relevant target properties via [get_target_property](https://cmake.org/cmake/help/v3.0/command/get_target_property.html).

File: CCDB/include/CCDB/Backend.h (hunk 1 above)
Comment (dennisklein) [inline CCDB/include/CCDB/Backend.h]: But your project got a [rule](https://htmlpreview.github.io/?https://github.com/AliceO2Group/CodingGuidelines/blob/master/coding_guidelines.html#Names_and_Order_of_Includes) for it anyways.

**Comment by ktf** [comment]
o2checkcode actually passed, which means that we did build O2 (and run the checker). I merge this so that other PRs can profit from it.

File: cmake/O2Dependencies.cmake (hunk 3 above)
Comment (Barthelemy) [inline cmake/O2Dependencies.cmake]: Hi Dennis, 
It is very nice that FairRoot now exports targets. Indeed the bucket system does not handle it yet. I will create an issue for this. I might come back to you if I have questions. 
Cheers,
https://alice.its.cern.ch/jira/browse/OCCF-31
